File: 357_numbers.js
#!/usr/bin/env node
// http://programmingpraxis.com/2015/03/06/357-numbers/

function threeFiveSevenNumber(n) {
  var t = n;
  while (t % 3 === 0) { t /= 3 };
  while (t % 5 === 0) { t /= 5 };
  while (t % 7 === 0) { t /= 7 };
  return t === 1 ? 1 : 0;
}

function find357NumbersBelow(limit) {
  var counter = 0
  for (var i = 1; i < limit; i += 2) { counter += threeFiveSevenNumber(i) }
  return counter
}

console.log(find357NumbersBelow(1000000))

File: 357_numbers.py
#!/usr/bin/env python
# http://programmingpraxis.com/2015/03/06/357-numbers/

def three_five_seven(number):
  while number % 3 == 0:
    number /= 3
  while number % 5 == 0:
    number /= 5
  while number % 7 == 0:
    number /= 7
  if number == 1:
    return 1
  else:
    return 0

acc = 0
for i in xrange(1,1000000,2):
  acc += three_five_seven(i)

print acc